lib/adobe/campaign/language.rb

```diff
@@ -0,0 +1,192 @@
+# frozen_string_literal: true
+module Adobe
+  module Campaign
+    class Language
+
+      def self.map_code(language_code)
+        return language_code if language_code.empty?
+
+        code = normalize language_code
+
+        return code if @adobe_language_codes.include?(code)
+        return code.gsub("-", "_") if @adobe_language_codes.include?(code.gsub("-", "_"))
+
+        if code.include?("-")
+          map_code code.rpartition("-").first
+        else
+          @adobe_language_codes.find{ |s| s.start_with? code}
+        end
+
+      end
+
+      def self.normalize(language_code)
+        code = language_code
+        code = code.downcase
+        code.gsub("_", "-")
```

Comment:
Why are you doing this if none of the `adobe_language_codes` have `-`s?

Reply:
Callers will most likely pass standard ISO format (i.e. '-' delimiter) and I wanted the code to be (somewhat) agnostic as regards Adobe Campaign format (such that the list could be updated with ISO format, or mixed).

I suppose I could 'Adobe normalize' first (i.e. change '-' to '_'), but the code will look much the same (only switching '-' with '_') that is, if I want it to remain (somewhat) agnostic about Adobe campaign formats supported.

Reply:
I've simplified the code such that it is no longer agnostic regarding Adobe language format.